js/search/downloader.js

```diff
@@ -313,57 +120,16 @@ const SubscriptionDownloader = new Lang.Class({
 
     fetch_update: function (subscription_id, cancellable, callback) {
         let task = new AsyncTask.AsyncTask(this, cancellable, callback);
-        task._subscription_id = subscription_id;
 
         task.catch_errors(() => {
-            this._app.hold();
-
-            if (this._ongoing_downloads.has(subscription_id)) {
-                task.return_value(false);
-                return;
-            }
-
-            this._ongoing_downloads.add(subscription_id);
-
-            let directory = this._subscriptions_dir.get_child(subscription_id);
-            Utils.ensure_directory(directory);
-
-            let manifest_file = directory.get_child('manifest.json');
-
-            let old_manifest = load_manifest(manifest_file, cancellable);
-            if (old_manifest === null)
-                old_manifest = { shards: [] };
-
-            let new_manifest_file = directory.get_child('manifest.json.new');
-            let uri = Format.vprintf('%s/v1/%s/manifest.json', [this._server_host, subscription_id]);
-            let manifest_uri = Soup.URI.new(uri);
-
-            let request = new FileDownloadRequest(manifest_uri, new_manifest_file);
-            this._file_downloader.download_file(request, cancellable, task.catch_callback_errors((source, download_task) => {
-                try {
-                    this._file_downloader.download_file_finish(download_task);
-                } catch(e if e.matches(Gio.ResolverError, Gio.ResolverError.NOT_FOUND)) {
-                    task.return_value(true);
-                    return;
-                }
-
-                let new_manifest = load_manifest(new_manifest_file, cancellable);
-                this._download_new_shards(directory, old_manifest, new_manifest, cancellable, task.catch_callback_errors((source, download_task) => {
-                    let have_new_manifest = this._download_new_shards_finish(download_task);
-                    if (!have_new_manifest)
-                        new_manifest_file.delete(cancellable);
-
-                    task.return_value(true);
-                }));
-            }));
+            let proxy = new DownloaderProxy(Gio.DBus.session, 'com.endlessm.EknSubscriptionsDownloader', '/com/endlessm/EknSubscriptionsDownloader');
+            proxy.DownloadSubscriptionRemote(subscription_id);
```

Comment:
This code will never call the callback, except if either of these statements throws an exception. If you want to call the callback, then you should add a `() => task.return_value()` to the `DownloadSubscriptionRemote` invocation here. But, it might be the case that the downloader should be fire-and-forget and the app has nothing sensible to do for error handling. In that case you could just remove the callback and the `fetch_update_finish` method altogether.